src/main/java/seedu/edrecord/storage/JsonAdaptedGrade.java

```diff
@@ -0,0 +1,91 @@
+package seedu.edrecord.storage;
+
+import java.util.Map;
+import java.util.Optional;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import seedu.edrecord.commons.exceptions.IllegalValueException;
+import seedu.edrecord.model.assignment.Grade;
+import seedu.edrecord.model.assignment.Score;
+import seedu.edrecord.model.assignment.Weightage;
+
+/**
+ * Jackson-friendly version of {@link Grade}.
```

Comment:
Not sure if it's fit to be called JsonAdaptedGrade, since `toModelType()` converts it to a `Map.Entry` object...?

Reply:
Created new issue #218. Will fix in next PR.